From 4043d4b7e2a6413cc4169d123b2e2a231d287d77 Mon Sep 17 00:00:00 2001
From: Danny Baumann <dannybaumann@web.de>
Date: Wed, 29 Aug 2018 11:04:58 +0200
Subject: [PATCH 14/15] soong_config: Allow process-specific override of target
 SDK version

Change-Id: I775e852c17af12d2233e644e3db10c10bcb20d19

# Conflicts:
#	build/soong/android/variable.go
#	build/soong/soong_config.mk
---
 build/soong/android/variable.go | 4 ++++
 build/soong/soong_config.mk     | 1 +
 2 files changed, 5 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index c09f6a18..a6da3dd5 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -25,6 +25,9 @@ type Product_variables struct {
 	}
 	Target_exfat_driver struct {
 		Cflags []string
+        }
+	Target_process_sdk_version_override struct {
+		Cppflags []string
 	}
 	Target_shim_libs struct {
 		Cppflags []string
@@ -55,6 +58,7 @@ type ProductVariables struct {
 	Needs_text_relocations  *bool `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
 	Target_exfat_driver  *string `json:",omitempty"`
+	Target_process_sdk_version_override *string `json:",omitempty"`
 	Target_shim_libs  *string `json:",omitempty"`
 	Uses_generic_camera_parameter_library  *bool `json:",omitempty"`
 	Uses_nvidia_enhancements  *bool `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index 3abe9955..f558c006 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -13,6 +13,7 @@ $(call add_json_str, Java_Source_Overlays, $(JAVA_SOURCE_OVERLAYS))
 $(call add_json_bool, Needs_text_relocations, $(filter true,$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS)))
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
 $(call add_json_str,  Target_exfat_driver,                   $(TARGET_EXFAT_DRIVER))
+$(call add_json_str_omitempty, Target_process_sdk_version_override, $(TARGET_PROCESS_SDK_VERSION_OVERRIDE))
 $(call add_json_str, Target_shim_libs, $(TARGET_LD_SHIM_LIBS))
 $(call add_json_bool, Uses_generic_camera_parameter_library, $(if $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY),,true))
 $(call add_json_bool, Uses_nvidia_enhancements, $(filter TRUE,$(NV_ANDROID_FRAMEWORK_ENHANCEMENTS)))
-- 
2.17.1

